### PR TITLE
extra set of wrappers and refactors

### PR DESCRIPTION
**Pull request by konstan**
- resource wrappers: module, data, infra
- refactored deployment, added data related bits
- added client-server chit-chat printout
- other refactors

### CODE REVIEW

**Comment by konstan** [comment]
@0xbase12 
The main part to look at are the changes in Api and Deployment modules as they are the ones that 
I was changing and they'd are mostly used everywhere else. There seems to be no regressions as this was validated to be working during the demo on job distributors for deployment state.
I didn't run any validations around NBs yet. I will do that when the code is in master and provide fixes if need be.

File: code/src/nuvla/api/api.py [hunk 1]
```diff
@@ -75,25 +75,71 @@
...
 
+class NuvlaResourceOperationNotAvailable(KeyError):
+    def __init__(self, reason, operation):
+        super().__init__(reason)
+        self.operation = operation
+
+
+def _request_debug(method, endpoint, params=None, doc=None,
+                   data=None, headers: Optional[dict]=None):
+    # TODO: Use logging library.
```
Comment (0xbase12) [inline code/src/nuvla/api/api.py:96]: @konstan There is a todo in the pull-request. Is it for later?

Comment (konstan) [reply code/src/nuvla/api/api.py:96]: @0xbase12 Yes, this is for later.

Comment (konstan) [reply code/src/nuvla/api/api.py:96]: @0xbase12 https://github.com/nuvla/python-library/issues/21

**Pull request merged**